src/main/kotlin/ecommerce/ui/CheckAdminInterceptor.kt

```diff
@@ -0,0 +1,29 @@
+package ecommerce.ui
+
+import ecommerce.auth.AuthorizationExtractor
+import ecommerce.auth.BearerAuthorizationExtractor
+import ecommerce.service.AuthService
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.web.servlet.HandlerInterceptor
+
+class CheckAdminInterceptor(
+    private val authService: AuthService,
+) : HandlerInterceptor {
+    private val authorizationExtractor: AuthorizationExtractor<String> = BearerAuthorizationExtractor()
+
+    override fun preHandle(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        handler: Any,
+    ): Boolean {
+        val token = authorizationExtractor.extract(request)
+        val member = authService.findMemberByToken(token)
+        if (member.role != "admin") {
+            response.status = 403
+            return false
+        }
+
+        return true
```

Comment:
Check the `org.springframework.http.HttpStatus`!
it's makes reduce human error 😄 

```suggestion
        val token = authorizationExtractor.extract(request)
        val member = authService.findMemberByToken(token)
        if (member.role != "admin") {
            response.status = HttpStatus.FORBIDDEN.value()
            return false
        }

        return true
```

Reply:
Thanks! Certainly it is much better!